tests/versioned/request-2.71.0/test_intercept.js

```diff
@@ -0,0 +1,5381 @@
+      t.end();
+    }).end();
+  }).end();
+});
+
+test('Persisted interceptors are in pendingMocks initially', function(t) {
+  var scope = nock('http://example.com')
+    .get('/abc')
+    .reply(200, "Persisted reply")
+    .persist();
+
+  t.deepEqual(scope.pendingMocks(), ["GET http://example.com:80/abc"]);
+  t.end();
+});
+
+test('Persisted interceptors are not in pendingMocks after the first request', function(t) {
+  var scope = nock('http://example.com')
+    .get('/def')
+    .reply(200, "Persisted reply")
+    .persist();
+
+  http.get('http://example.com/def', function(res) {
+    t.deepEqual(scope.pendingMocks(), []);
+    t.end();
+  });
+});
+
+test("persist reply with file", function(t) {
+  nock('http://www.filereplier.com')
+    .persist()
+    .get('/')
+    .replyWithFile(200, __dirname + '/../assets/reply_file_1.txt')
+    .get('/test')
+    .reply(200, 'Yay!');
+
+  async.each([1,2], function(_, cb) {
+    var dataCalled = false;
+    var req = http.request({
+        host: "www.filereplier.com"
+      , path: '/'
+      , port: 80
+    }, function(res) {
+
+      t.equal(res.statusCode, 200);
+      res.once('end', function() {
+        t.ok(dataCalled);
+        cb();
+      });
+      res.on('data', function(data) {
+        dataCalled = true;
+        t.equal(data.toString(), "Hello from the file!", "response should match");
+      });
+
+    });
+    req.end();
+  }, t.end.bind(t));
+});
+
+test('(re-)activate after restore', {skip: process.env.AIRPLANE}, function(t) {
+  var scope = nock('http://google.com')
+    .get('/')
+    .reply(200, 'Hello, World!');
+
+  nock.restore();
+  t.false(nock.isActive());
+
+  http.get('http://google.com/', function(res) {
+    res.resume();
+    res.on('end', function() {
+      t.ok(!scope.isDone());
+
+      nock.activate();
+      t.true(nock.isActive());
+      http.get('http://google.com', function(res) {
+        res.resume();
+        res.on('end', function() {
+          t.ok(scope.isDone());
+          t.end();
+        });
+      }).end();
+    });
+  }).end();
+});
+
+test("allow unmocked option works with https", {skip: process.env.AIRPLANE}, function(t) {
+  t.plan(5)
+  var scope = nock('https://www.google.com', {allowUnmocked: true})
+    .get('/abc')
+    .reply(200, 'Hey!')
+    .get('/wont/get/here')
+    .reply(200, 'Hi!');
+
+  function secondIsDone() {
+    t.ok(! scope.isDone());
+    https.request({
+        host: "www.google.com"
+      , path: "/"
+    }, function(res) {
+      res.resume();
+      t.ok(true, 'Google replied to /');
+      res.destroy();
+      t.assert(res.statusCode < 400 && res.statusCode >= 200, 'GET Google Home page');
+    }).end();
+  }
+
+  function firstIsDone() {
+    t.ok(! scope.isDone(), 'scope is not done');
+    https.request({
+        host: "www.google.com"
+      , path: "/does/not/exist/dskjsakdj"
+    }, function(res) {
+      t.equal(404, res.statusCode, 'real google response status code');
+      res.on('d
```

Comment:
Is this the same as `tests/test_intercept.js`? If yes, couldn't we load that instead?